cluster-autoscaler/cloudprovider/huaweicloud/huaweicloud_service_manager.go

```diff
@@ -502,6 +502,29 @@ func (csm *cloudServiceManager) getScalingGroupByID(groupID string) (*huaweiclou
 	return response.ScalingGroup, nil
 }
 
+func (csm *cloudServiceManager) listScalingTagsByID(groupID string) (map[string]string, error) {
+	asClient := csm.getASClientFunc()
+	opts := &huaweicloudsdkasmodel.ListScalingTagInfosByResourceIdRequest{
+		ResourceType: huaweicloudsdkasmodel.GetListScalingTagInfosByResourceIdRequestResourceTypeEnum().SCALING_GROUP_TAG,
+		ResourceId:   groupID,
+	}
+	response, err := asClient.ListScalingTagInfosByResourceId(opts)
+	if err != nil {
+		klog.Errorf("failed to list scaling group tags. scaling group id: %s, error: %v", groupID, err)
+		return nil, err
+	}
+	if response == nil || response.Tags == nil {
+		return nil, nil
+	}
+
+	tags := map[string]string{}
+	for _, tag := range *response.Tags {
+		tags[tag.Key] = *tag.Value
```

Comment:
PANIC RISK.
AS group allows people to add tags with an `empty` value, then the `tag.Value` will be nil.
We must check tag.Value here. like :
```
		if tag.Value != nil {
			tags[tag.Key] = *tag.Value
			continue
		}

		// the tag key of the Auto Scaling group might be empty
		tags[tag.Key] = ""
```

Reply:
I test, it will return empty string:
```json
{
 "tags": [
  {
   "key": "huawei",
   "value": ""
  }
 ],
 "sys_tags": null
}
```
So I think it's not necessary to make it complex